cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go

```diff
@@ -406,16 +406,18 @@ func TestNodeGroupIncreaseSize(t *testing.T) {
 
 func TestNodeGroupDecreaseTargetSize(t *testing.T) {
 	type testCase struct {
-		description                        string
-		delta                              int
-		initial                            int32
-		targetSizeIncrement                int32
-		expected                           int32
-		expectedError                      bool
-		includeDeletingMachine             bool
-		includeFailedMachine               bool
-		includeFailedMachineWithProviderID bool
-		includePendingMachine              bool
+		description                         string
+		delta                               int
+		initial                             int32
+		targetSizeIncrement                 int32
+		expected                            int32
+		expectedError                       bool
+		includeDeletingMachine              bool
+		includeFailedMachine                bool
+		includeFailedMachineWithProviderID  bool
+		includePendingMachine               bool
+		includePendingMachineWithProviderID bool
```

Comment:
just a note that these UT scenarios could probably be improved in the future by allowing a more flexible count of machines with particular states (right now we only organize our scenarios so they have one of each type of [not sure about the combinatorics] failed/pending/no providerID)

Reply:
yes, what i want to do as a next step is to change our unit test infrastructure to have a more fluent style builder for the capi objects. i think that will make the tests easier to understand. this is a middle ground.

Reply:
as to the combinatorics, i have tried to make it so that we can test combinations of machines. i would like to add that during the next step.